File: rust/src/bin/ziyy.rs
use std::env;
use std::fs::File;
use std::io::{stdout, BufReader, Read, Write};
use std::path::Path;
use std::process::exit;
use ziyy::Compiler;

pub fn compile(source: &str, out: &mut impl Write) -> ziyy::Result<()> {
    let mut compiler = Compiler::new(source, out, None);
    compiler.compile()?;
    Ok(())
}

fn usage() {
    let mut out = stdout().lock();
    compile(include_str!("../../../help.zi"), &mut out).unwrap();
}

fn main() {
    let mut args = env::args();
    let mut out: Vec<u8> = vec![];
    if args.len() - 1 < 1 {
        usage();
        exit(0);
    }
    let first = args.nth(1);
    if first == Some("-n".to_string()) || first == Some("--no-newline".to_string()) {
        if compile(args.next().unwrap().as_str(), &mut out).is_err() {
            exit(1);
        }
    } else if first == Some("-f".to_string()) || first == Some("--file".to_string()) {
        if args.len() == 0 {
            usage();
            exit(1);
        }
        let file = args.next().unwrap();
        if !Path::new(&file).is_file() {
            usage();
            exit(1);
        }
        let f = File::open(file).unwrap();
        let mut reader = BufReader::new(f);
        let mut file = String::new();
        let _ = reader.read_to_string(&mut file);
        if compile(file.as_str(), &mut out).is_err() {
            exit(1)
        }
    } else if first == Some("-V".to_string()) || first == Some("--version".to_string()) {
        println!("ziyy 2.0.0-beta.2")
    } else if first == Some("-h".to_string()) || first == Some("--help".to_string()) {
        usage();
        exit(0);
    } else {
        if compile(first.unwrap().as_str(), &mut out).is_err() {
            exit(1)
        }
        let _ = writeln!(out);
        let _ = stdout().write(&out);
    }
}

File: javascript/src/ziyy.ts
import { readFileSync, statSync } from "fs";
import { compile } from "./style";

const out = process.stdout

function usage() {
    compile(`Convenient Terminal Output Styler.

<green><b><u>Usage:</u></b> <cyan><b>ziyy</b> <i>[OPTION] [TEXT]</i></cyan>

<b><u>Options:</u></b></green>
  <cyan><b>-V</b></cyan>, <cyan><b>--version</b></cyan>
          Print version info and exit
  <cyan><b>-f</b></cyan>, <cyan><b>--file</b> \\<FILENAME\\></cyan>
          Read input from file.
  <cyan><b>-n</b></cyan>, <cyan><b>--no-newline</b></cyan>
          Do not print newline after text.
  <cyan><b>-h</b></cyan>, <cyan><b>--help</b></cyan>
          Print help
`, out)
}

const args = process.argv.slice(2)
if (args.length < 1) {
    usage()
    process.exit(0)
}
const first = args[0]
if (first == "-n" || first == "--no-newline")
    compile(args[1], out)
else if (first == "-f" || first == "--file") {
    if (args.length == 1)
        process.exit(1)
    let file = statSync(args[1])
    if (!file.isFile())
        process.exit()
    compile(readFileSync(args[1], "utf8"), out)
}
else if (first == "-V" || first == "--version") {
    console.log("ziyy 1.0.6")
}
else if (first == "-h" || first == "--help") {
    usage()
    process.exit()
}
else {
    compile(first, out)
}